knowledge-base/scheduler-conditionally-disable-drag-for-some-appointments.md

```diff
@@ -0,0 +1,151 @@
+        <SchedulerWeekView StartTime="@DayStart" />
+        <SchedulerMonthView />
+    </SchedulerViews>
+</TelerikScheduler>
+
+@code {
+    private DateTime StartDate { get; set; } = new DateTime(2019, 11, 29);
+
+    private SchedulerView CurrView { get; set; } = SchedulerView.Week;
+
+    private DateTime DayStart { get; set; } = new DateTime(2000, 1, 1, 8, 0, 0);
+
+    private void OnItemRenderHandler(SchedulerItemRenderEventArgs e)
+    {
+        SchedulerAppointment appt = e.Item as SchedulerAppointment;
+        
+        //add custom class to stop the pointer events of the appointment
+        if (appt.NonDraggable)
+        {
+            e.Class = "non-draggable-appointment";
+        }
+    }
+
+    private void OnUpdate(SchedulerUpdateEventArgs args)
+    {
+        SchedulerAppointment item = (SchedulerAppointment)args.Item;
+
+        if (item.NonDraggable)
+        {
+            //Do not execute code for updating the appointment.
+        }
+        else
+        {
+            //Execute appointment update.
+            //The actual update is not handled here for brevity.
+        }
+    }
+
+    private List<SchedulerAppointment> Appointments = new List<SchedulerAppointment>()
+    {
+            new SchedulerAppointment
+            {
+                Title = "Vet visit",
+                Description = "The cat needs vaccinations and her teeth checked.",
+                Start = new DateTime(2019, 11, 26, 11, 30, 0),
+                End = new DateTime(2019, 11, 26, 12, 0, 0),
+                NonDraggable = true
+            },
+
+            new SchedulerAppointment
+            {
+                Title = "Planning meeting",
+                Description = "Kick off the new project.",
+                Start = new DateTime(2019, 11, 25, 9, 30, 0),
+                End = new DateTime(2019, 11, 25, 12, 45, 0)
+            },
+
+            new SchedulerAppointment
+            {
+                Title = "Trip to Hawaii",
+                Description = "An unforgettable holiday!",
+                IsAllDay = true,
+                Start = new DateTime(2019, 11, 29),
+                End = new DateTime(2019, 12, 07),
+                NonDraggable = true
+            },
+
+            new SchedulerAppointment
+            {
+                Title = "Online conference",
+                IsAllDay = true,
+                Start = new DateTime(2019, 11, 25),
+                End = new DateTime(2019, 11, 26)
+            }
+    };
+
+    public class SchedulerAppointment
+    {
+        public string Title { get; set; }
+        public string Description { get; set; }
+        public DateTime Start { get; set; }
+        public DateTime End { get; set; }
+        public bool IsAllDay { get; set; }
+
+        public bool NonDraggable { get; set; }
+    }
+}
+
+<style>
+    .non-draggable-appointment {
+        background-color: grey;
+        font-weight: 900;
+        pointer-events: none !important;
+    }
+</style>
```

Comment:
Move the CSS code above the `@code` section.

Check if it's possible to restore pointer events to the appointment's close button.

Reply:
Updated in https://github.com/telerik/blazor-docs/pull/2297/commits/ab625b4c33c47ae03e7e67f8bc5b88b45ed4bf50